storage/cloudimpl/cloudimpltests: TestHttpGet failed

[(storage/cloudimpl/cloudimpltests).TestHttpGet failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2061897&tab=buildLog) on [master@b6e7102a76b696bde6acd9d5d0ffcccfbf37006b](https://github.com/cockroachdb/cockroach/commits/b6e7102a76b696bde6acd9d5d0ffcccfbf37006b):

```
E200703 06:29:45.011642 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 57)
E200703 06:29:45.023616 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 58)
E200703 06:29:45.034358 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": read tcp 127.0.0.1:46884->127.0.0.1:34731: read: connection reset by peer (attempt 59)
E200703 06:29:45.045145 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 60)
E200703 06:29:45.057081 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 61)
E200703 06:29:45.067813 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 62)
E200703 06:29:45.080717 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 63)
E200703 06:29:45.092190 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 64)
E200703 06:29:45.102753 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 65)
E200703 06:29:45.113304 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 66)
E200703 06:29:45.126640 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": read tcp 127.0.0.1:46900->127.0.0.1:34731: read: connection reset by peer (attempt 67)
E200703 06:29:45.137094 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 68)
E200703 06:29:45.149161 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 69)
E200703 06:29:45.160694 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 70)
E200703 06:29:45.171437 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 71)
E200703 06:29:45.184379 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 72)
E200703 06:29:45.197033 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 73)
E200703 06:29:45.209158 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 74)
E200703 06:29:45.220975 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 75)
E200703 06:29:45.233110 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 76)
E200703 06:29:45.243683 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 77)
E200703 06:29:45.253962 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 78)
E200703 06:29:45.264770 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 79)
E200703 06:29:45.277646 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 80)
E200703 06:29:45.288425 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 81)
E200703 06:29:45.300691 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 82)
E200703 06:29:45.313069 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 83)
E200703 06:29:45.324152 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 84)
E200703 06:29:45.334891 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 85)
E200703 06:29:45.346597 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 86)
E200703 06:29:45.359643 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 87)
E200703 06:29:45.371607 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 88)
E200703 06:29:45.382538 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 89)
E200703 06:29:45.395556 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": read tcp 127.0.0.1:46946->127.0.0.1:34731: read: connection reset by peer (attempt 90)
E200703 06:29:45.406408 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 91)
E200703 06:29:45.417644 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 92)
E200703 06:29:45.430526 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 93)
E200703 06:29:45.442402 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 94)
E200703 06:29:45.455244 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": read tcp 127.0.0.1:46956->127.0.0.1:34731: read: connection reset by peer (attempt 95)
E200703 06:29:45.466373 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 96)
E200703 06:29:45.478121 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 97)
E200703 06:29:45.491448 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 98)
E200703 06:29:45.502161 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": http: server closed idle connection (attempt 99)
E200703 06:29:45.513057 6436 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:34731/something": EOF (attempt 100)
    TestHttpGet/read-2: http_storage_test.go:268: 
        	Error Trace:	http_storage_test.go:268
        	Error:      	Received unexpected error:
        	            	retryable http error: Get "http://127.0.0.1:34731/something": EOF
        	Test:       	TestHttpGet/read-2
    --- FAIL: TestHttpGet/read-2 (1.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestHttpGet PKG=./pkg/storage/cloudimpl/cloudimpltests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestHttpGet.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage/cloudimpl/cloudimpltests).TestHttpGet failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2073705&tab=buildLog) on [master@e3fb5aa18d0f5064f7ba5d4df3864e94b3abb96d](https://github.com/cockroachdb/cockroach/commits/e3fb5aa18d0f5064f7ba5d4df3864e94b3abb96d):

```
E200708 07:54:55.633118 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 57)
E200708 07:54:55.645592 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 58)
E200708 07:54:55.658633 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 59)
E200708 07:54:55.669870 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 60)
E200708 07:54:55.682902 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 61)
E200708 07:54:55.694573 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 62)
E200708 07:54:55.707070 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 63)
E200708 07:54:55.720141 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33666->127.0.0.1:36531: read: connection reset by peer (attempt 64)
E200708 07:54:55.730867 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 65)
E200708 07:54:55.743509 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33678->127.0.0.1:36531: read: connection reset by peer (attempt 66)
E200708 07:54:55.754298 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 67)
E200708 07:54:55.765911 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 68)
E200708 07:54:55.778158 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 69)
E200708 07:54:55.790050 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 70)
E200708 07:54:55.803041 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 71)
E200708 07:54:55.816256 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 72)
E200708 07:54:55.827485 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 73)
E200708 07:54:55.838613 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 74)
E200708 07:54:55.850331 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 75)
E200708 07:54:55.862491 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 76)
E200708 07:54:55.873063 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 77)
E200708 07:54:55.884340 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 78)
E200708 07:54:55.894729 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 79)
E200708 07:54:55.906044 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 80)
E200708 07:54:55.918448 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 81)
E200708 07:54:55.932068 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 82)
E200708 07:54:55.944772 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33838->127.0.0.1:36531: read: connection reset by peer (attempt 83)
E200708 07:54:55.956005 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 84)
E200708 07:54:55.967526 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 85)
E200708 07:54:55.979086 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 86)
E200708 07:54:55.991849 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 87)
E200708 07:54:56.003703 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33876->127.0.0.1:36531: read: connection reset by peer (attempt 88)
E200708 07:54:56.016635 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33890->127.0.0.1:36531: read: connection reset by peer (attempt 89)
E200708 07:54:56.027158 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 90)
E200708 07:54:56.038180 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 91)
E200708 07:54:56.049538 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 92)
E200708 07:54:56.060306 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 93)
E200708 07:54:56.071080 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 94)
E200708 07:54:56.082900 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 95)
E200708 07:54:56.093582 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 96)
E200708 07:54:56.107214 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33924->127.0.0.1:36531: read: connection reset by peer (attempt 97)
E200708 07:54:56.118406 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": http: server closed idle connection (attempt 98)
E200708 07:54:56.131925 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": read tcp 127.0.0.1:33940->127.0.0.1:36531: read: connection reset by peer (attempt 99)
E200708 07:54:56.142414 9525 storage/cloudimpl/http_storage.go:202  HTTP:Req error: err=retryable http error: Get "http://127.0.0.1:36531/something": EOF (attempt 100)
    TestHttpGet/read-5: http_storage_test.go:268: 
        	Error Trace:	http_storage_test.go:268
        	Error:      	Received unexpected error:
        	            	retryable http error: Get "http://127.0.0.1:36531/something": EOF
        	Test:       	TestHttpGet/read-5
    --- FAIL: TestHttpGet/read-5 (1.33s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestHttpGet PKG=./pkg/storage/cloudimpl/cloudimpltests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestHttpGet.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/storage/cloudimpl/cloudimpltests/http_storage_test.go
+++ b/pkg/storage/cloudimpl/cloudimpltests/http_storage_test.go
@@ -193,9 +193,13 @@ func TestHttpGet(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	data := []byte("to serve, or not to serve.  c'est la question")
 
+	defer func(opts retry.Options) {
+		cloudimpl.HTTPRetryOptions = opts
+	}(cloudimpl.HTTPRetryOptions)
+
 	cloudimpl.HTTPRetryOptions.InitialBackoff = 1 * time.Microsecond
 	cloudimpl.HTTPRetryOptions.MaxBackoff = 10 * time.Millisecond
-	cloudimpl.HTTPRetryOptions.MaxRetries = 100
+	cloudimpl.HTTPRetryOptions.MaxRetries = 25
 
 	for _, tc := range []int{1, 2, 5, 16, 32, len(data) - 1, len(data)} {
 		t.Run(fmt.Sprintf("read-%d", tc), func(t *testing.T) {
@@ -234,7 +238,7 @@ func TestHttpGet(t *testing.T) {
 			g.GoCtx(func(ctx context.Context) error {
 				opts := retry.Options{
 					InitialBackoff: 500 * time.Microsecond,
-					MaxBackoff:     1 * time.Millisecond,
+					MaxBackoff:     5 * time.Millisecond,
 				}
 				for attempt := retry.StartWithCtx(ctx, opts); attempt.Next(); {
 					s.CloseClientConnections()

